browser-test/src/trusted_intermediary.test.ts

```diff
@@ -108,6 +108,108 @@ describe('Trusted intermediaries', () => {
 
     await validateScreenshot(page, 'dashboard-add-clients-no-email')
   })
+  it('expect client email address to be updated', async () => {
+    const {page, tiDashboard} = ctx
+    await loginAsTrustedIntermediary(page)
+    await tiDashboard.gotoTIDashboardPage(page)
+    await waitForPageJsLoad(page)
+    const client: ClientInformation = {
+      emailAddress: 'test@sample.com',
+      firstName: 'first',
+      middleName: 'middle',
+      lastName: 'last',
+      dobDate: '2021-06-10',
+    }
+    await tiDashboard.createClient(client)
+    await waitForPageJsLoad(page)
+    await tiDashboard.updateClientEmailAddress(client, 'new@email.com')
+
+    const updatedClient: ClientInformation = {
+      emailAddress: 'new@email.com',
+      firstName: 'first',
+      middleName: 'middle',
+      lastName: 'last',
+      dobDate: '2021-06-10',
+    }
+    await tiDashboard.expectDashboardContainClient(updatedClient)
+  })
+  it('expect client ti notes and phone to be updated', async () => {
+    const {page, tiDashboard} = ctx
+    await loginAsTrustedIntermediary(page)
+    await tiDashboard.gotoTIDashboardPage(page)
+    await waitForPageJsLoad(page)
+    const client: ClientInformation = {
+      emailAddress: 'test@sample.com',
+      firstName: 'first',
+      middleName: 'middle',
+      lastName: 'last',
+      dobDate: '2021-06-10',
+    }
+    await tiDashboard.createClient(client)
+    await waitForPageJsLoad(page)
+    await tiDashboard.updateClientTiNoteAndPhone(
+      client,
+      'Housing Assitance',
+      '4256007121',
+    )
+    await waitForPageJsLoad(page)
+    await tiDashboard.expectClientContainsTiNoteAndPhone(
+      client,
+      'Housing Assitance',
+      '4256007121',
+    )
+    await validateScreenshot(page, 'edit-client-information-with-all-fields')
+  })
+  it('expect back button to land in dashboard in the edit client page', async () => {
+    const {page, tiDashboard} = ctx
+    await loginAsTrustedIntermediary(page)
+    await tiDashboard.gotoTIDashboardPage(page)
+    await waitForPageJsLoad(page)
+    const client: ClientInformation = {
+      emailAddress: 'tes@sample.com',
+      firstName: 'first',
+      middleName: 'middle',
+      lastName: 'last',
+      dobDate: '2021-06-10',
+    }
+    await tiDashboard.createClient(client)
+    await waitForPageJsLoad(page)
+    await page
+      .getByRole('row')
+      .filter({hasText: client.emailAddress})
+      .getByText('Edit')
+      .click()
+    await waitForPageJsLoad(page)
+    await page.waitForSelector('h2:has-text("Edit Client")')
+    await page.click('text=Back to client list')
+    await waitForPageJsLoad(page)
+    await validateScreenshot(page, 'back-link-leads-to-ti-dashboard')
```

Comment:
This is a very simple test to check if the back button is redirecting to the dashboard. Snapshots has worked fine in the past. But I can extend it to check for some text content.

Reply:
Austin gave me this feedback on a PR recently: https://github.com/civiform/civiform/pull/5752

I think checking for text is preferred because failures will show up faster.